theano/gpuarray/opt.py

```diff
@@ -845,10 +848,14 @@ def local_gpu_pdbbreakpoint_op(node):
 def local_gpua_lazy_ifelse(op, context_name, inputs):
     if op.gpu:
         return
+    if isinstance(inputs[0].type, GpuArrayType):
+        return
```

Comment:
The condition is always on the CPU, so it should never be GpuArrayType. So those 2 lines are useless.